apps/web/lib/supabase/server.ts

```diff
@@ -7,8 +8,8 @@ export function createClient(): SupabaseClient<Database> {
   const cookieStore = cookies()
 
   return createServerClient<Database>(
-    process.env.NEXT_PUBLIC_SUPABASE_URL!,
-    process.env.NEXT_SUPABASE_KEY!,
+    env.NEXT_PUBLIC_SUPABASE_URL,
+    env.NEXT_PUBLIC_SUPABASE_ANON_KEY,
```

Comment:
This key is not public, and not an anon key from supabase.

So the previous name are better, to next not expose it, and to represent the right key on supabase, please revert it to `NEXT_SUPABASE_KEY`

Reply:
Next environment variables must have the prefix `NEXT_PUBLIC`, we will get type errors if these are not prefixed.

We can put `NEXT_PUBLIC_SUPABASE_KEY`

Reply:
`NEXT_PUBLIC` is when you want to expose this envs to the browser.

In this case, this env cannot be public, because is really sensitive data.

So because of that it not have `NEXT_PUBLIC` prefix, just `NEXT`, to we know this is used on Vercel cloud.

I running it with dev command
![Screenshot 2024-06-02 at 20 18 31](https://github.com/brunocroh/bolhadev.chat/assets/13812512/72e71be5-bdfb-4572-81cf-10489f59efe3)

and running build
![Screenshot 2024-06-02 at 20 19 00](https://github.com/brunocroh/bolhadev.chat/assets/13812512/d0300d90-a52f-4c50-8a3d-c5b2ee920e4e)

and not have any warnings about it on my side.
If I skipping something let me know please.

But we cannot add `NEXT_PUBLIC` on this one, the env `NEXT_SUPABASE_KEY` is just consumed on server action.

Reply:
I got it, thanks.

I was putting them as client env so I was getting a warning. I put them as server env and it worked

Reply:
Great!